File: insertionsort.py
def insertionSort(arr: list[int]) -> list[int]:
  """
    Insertion sort algorithm that performs sorting of lists through a predetermined set of procedures.

    # Procedures
    - It starts its operations from the second element in the list and checks its value with the element before it.
    - If its value is less than the element, then the values are swapped.
    - However, if the value is greater than the element, then nothing happens to the list.
    - Finally, we move to the next element in the list and repeat the procedures.
  """
  for i in range(1, len(arr)):
    while arr[i - 1] > arr[i] and i > 0:
      arr[i - 1], arr[i] = arr[i], arr[i - 1]
      i -= 1

  print(arr)


insertionSort([2,1,3,6,4,5])

File: insertionsort.cpp
#include <iostream>
#include <algorithm> // For std::swap

void insertionSort(int* num, int size) {
  for (int i = 1; i < size; i++) {
    int j = i;
    while (j > 0 && num[j - 1] > num[j]) {
      std::swap(num[j - 1], num[j]);
      j--;
    }
  }
}

int main() {
  int arr[] = {2, 1, 3, 5, 4, 6, 11, 50, 8, 10};
  int size = sizeof(arr) / sizeof(arr[0]);  // Calculate array size

  insertionSort(arr, size);

  // Print sorted array
  for (int i = 0; i < size; i++) {
    std::cout << arr[i] << " ";
  }
  std::cout << std::endl;

  return 0;
}
